peripherals/adc/adc_v3_g4.yaml

```diff
@@ -0,0 +1,95 @@
+# ADC v3 with G4 specific fields
+
+_include:
+  - "adc_v3.yaml"
+
+"ADC,ADC?":
+  DIFSEL:
+    _merge:
+      - "DIFSEL*"
+  CR:
+    ADVREGEN:
+      Intermediate: [0, "Intermediate state required when moving the ADC voltage regulator between states"]
+      Enabled: [1, "ADC voltage regulator enabled"]
+      Disabled: [2, "ADC voltage regulator disabled"]
+    DEEPPWD:
+      Disabled: [0, ADC not in Deep-power down]
+      Enabled: [1, ADC in Deep-power-down (default reset state)]
+
+  CFGR:
+    JQDIS:
+      Enabled: [0, Injected Queue enabled]
+      Disabled: [1, Injected Queue disabled]
+    AWD1CH: [0, 18]
+    EXTSEL:
+      HRTIM_ADCTRG1: [7, "HRTIM_ADCTRG1 event"]
+      HRTIM_ADCTRG3: [8, "HRTIM_ADCTRG3 event"]
+    ALIGN:
+      Right: [0, "Right alignment"]
+      Left: [1, "Left alignment"]
+    RES:
+      Bits12: [0, "12-bit"]
+      Bits10: [1, "10-bit"]
+      Bits8: [2, "8-bit"]
+      Bits6: [3, "6-bit"]
+    DMACFG:
+      OneShot: [0, "DMA One Shot Mode selected"]
+      Circular: [1, "DMA circular mode selected"]
+    DMAEN:
+      Disabled: [0, "DMA disabled"]
+      Enabled: [1, "DMA enabled"]
+  "CFGR2":
+    SMPTRIG:
+      Disabled: [0, Sampling time not controlled by trigger]
+      Enabled: [1, Sampling time controlled by hardware or software trigger]
+    BULB:
+      Disabled: [0, Bulb sampling mode disabled]
+      Enabled: [1, "Bulb sampling mode enabled. Immediately start sampling after last conversion finishes."]
+    SWTRIG:
+      Disabled: [0, End sampling period and start conversion]
+      Enabled: [1, Start sampling period]
+    SMPTRIG:
+      Disabled: [0, Sampling time not controlled by HW or SW trigger]
+      Enabled: [1, Sampling time controlled by HW or SW trigger]
+    GCOMP:
+      Disabled: [0, Regular ADC operating mode]
+      Enabled: [1, Gain compensation enabled and applies to all channels]
+    ROVSM:
+      Continued: [0, "Oversampling is temporary stopped and continued after injection sequence"]
+      Resumed: [1, "Oversampling is aborted and resumed from start after injection sequence"]
+    TROVS:
+      Automatic: [0, "All oversampled conversions for a channel are run following a trigger"]
+      Triggered: [1, "Each oversampled conversion for a channel needs a new trigger"]
+    OVSS: [0, 11]
+    OVSR: [0, 7]
```

Comment:
As with `OVSS` I think it would be clearer to have `OS2`, `OS4`, etc enumerations.